File: bs.js
const binarySearch = (arr, target) => {
  left = 0
  right = arr.length -1
  mid = 0

  while (left<= right) {
    mid = Math.floor((left + right) / 2)
    if (arr[mid] > target) right = mid - 1
    else if (arr[mid] < target) left = mid + 1
    else return `Element is present at index ${mid}`
  }
  return 'Element does not exist!'
}




const list = [1,2,3,4,5,6,7,8,9,10]
const target = 10
let bs = binarySearch(list, target)
console.log(bs)

File: bs.py
def binary_search(lst, target):

    left_index = 0
    right_index = len(lst) - 1
    mid_index = 0

    while left_index <= right_index:
        mid_index = (left_index + right_index) // 2

        if lst[mid_index] > target:
            right_index = mid_index - 1

        elif lst[mid_index] < target:
            left_index = mid_index + 1
        else:
            return 'Element is found at index ' + str(mid_index)
    return 'Element does not exist!'


lst = list(range(1, 100))
target = 77
bs = binary_search(lst, target)
print(bs)
